src/libData/AccountData/Account.cpp

```diff
@@ -59,17 +64,22 @@ void Account::InitStorage() {
   }
 }
 
-void Account::InitContract(const bytes& data, const Address& addr) {
+bool Account::InitContract(const bytes& data, const Address& addr) {
   SetInitData(data);
-  InitContract(addr);
+  m_address = addr;
```

Comment:
m_address = addr;

set m_address after `if (!InitContract(addr))` to prevent any inconsistent state

Reply:
fixed